Source/ACE.Common/Cryptography/CryptoSystem.cs

```diff
@@ -8,10 +8,10 @@ namespace ACE.Common.Cryptography
 {
     /// <summary>
     /// Forward looking cache for ISAAC calculations.<para />
-    /// Manages ISAAC stream cipher and key searches.<para />
+    /// Manages ISAAC stream cipher and performs key searches.<para />
     /// Manages a range of encryption keys generated by the stream cipher.<para />
     /// </summary>
-    public class CryptoSystem
+    public class CryptoSystem : IDisposable
```

Comment:
Only change requested would be to remove IDisposable. See my comment on the main PR conversation.

Reply:
I'm sorry but that's wrong.  The IDisposable pattern absolutely works for managed objects and the habit of cleaning up reference objects in the **managed section** of the IDisposable pattern is encouraged and follows Microsoft best practices.   [It is highly inefficient for an application to force garbage collection as part of the service control flow.](https://lostechies.com/chrispatterson/2012/11/29/idisposable-done-right/)

Reply:
In the link you provide, he also references MS documentation that is titled "Implementing Finalize and Dispose to Clean Up **Unmanaged** Resources"

In this case, we're talking about implementing excessive cleanup for single EventHandler subscription per Session/NetworkSession, aka, 1 per client connection.

When a Session drops, the tie to root will be lost, and GC will clean these up. I also know this because I've spent considerable amount of time analyzing memory profiles from very populated ACE servers in effort to help reduce memory consumption.

Reply:
i would like a second opinion

Reply:
> In this case, we're talking about implementing excessive cleanup for single EventHandler subscription per Session/NetworkSession, aka, 1 per client connection.

you are wrong